service/create/policy.go

```diff
@@ -102,3 +116,69 @@ func createInstanceProfile(svc *iam.IAM, clusterID string) (string, error) {
 
 	return clusterProfileName, nil
 }
+
+func deleteInstanceProfile(svc *iam.IAM, clusterID string) error {
+	clusterProfileName := fmt.Sprintf("%s-%s", clusterID, ProfileNameTemplate)
+
+	_, err := svc.DeleteInstanceProfile(&iam.DeleteInstanceProfileInput{
+		InstanceProfileName: aws.String(clusterProfileName),
+	})
+
+	if err != nil {
+		return err
```

Comment:
@xh3b4sd true, I noticed after opening the PR.

I didn't mask in this file because we don't import `microerror` there and I didn't want to pollute the namespace, so I  wrapped the error in [the caller](https://github.com/kinvolk/aws-operator/blob/1f8780bb649532b1f62049298d57c4ce34e129ab/service/create/service.go#L273). The output is less informative (the line printed is not where the error _really_ happened) but the error message should be clear enough.

WDYT?